File: test/test_rcorr.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <mkl.h>
#include <memory.h>
#include "params.h"
#include "integration.h"
#include "constants.h"
#include "matsubara.h"
#include "save.h"
#include <time.h>
#include "util.h"

void test_rcorr()
{
  printf("\n---------------------------------\n");
  printf("test rcorr weights\n---------------------------------\n");

  cdouble * rcorr = (cdouble *)MKL_malloc(order*order*order*sizeof(cdouble), MEM_DATA_ALIGN);

  char dsetname[32];
  
  sprintf(dsetname, "/rcorr_p%d", order);

  printf("dset name = %s\n", dsetname);

  dload("weights.h5", dsetname, rcorr, order*order*order);

  zsave("../test/results/test_rcorr.h5", "/rcorr", rcorr, order*order*order);

  system("exec python3 ../test/test_rcorr.py");
}

File: test/test_rcorr.py
import time
import numpy as np
import h5py
from read_params import read_params
from util import *

dr = np.load("../bin/rcorr.npy").item();
rcorr = dr[6]
order = np.shape(rcorr)[0]
rcorr = np.reshape(rcorr, [-1])
rcorr = np.reshape(rcorr, [order, order, order], order='F')

crcorr = zload("../test/results/test_rcorr.h5", "/rcorr")
crcorr = np.reshape(crcorr, [order, order, order], order='F')

print('diff = {}'.format(np.mean(np.abs(crcorr - rcorr))))



